### PR TITLE
core/finality-grandpa: Minor refactorings

### PR DESCRIPTION
**Pull request by mxinden**
Follow up to https://github.com/paritytech/substrate/pull/3755/ doing minor refactorings within `core/finality-grandpa`. See individual commit messages for details.

### CODE REVIEW

File: core/finality-grandpa/src/communication/mod.rs [hunk 1]
```diff
@@ -202,7 +211,15 @@ impl<B, S, H> Network<B> for Arc<NetworkService<B, S, H>> where
...
+///
+/// 1. polling the oneshot until it returns a message channel
+///
+/// 2. polling the message channel for messages
+///
+/// `NetworkStream` combines the two steps into one, requiring a consumer to only poll `NetworkStream` to retrieve
+/// messages directly.
 pub struct NetworkStream {
 	inner: Option<mpsc::UnboundedReceiver<network_gossip::TopicNotification>>,
 	outer: oneshot::Receiver<mpsc::UnboundedReceiver<network_gossip::TopicNotification>>
```
Comment (thiolliere) [inline core/finality-grandpa/src/communication/mod.rs]: note: this struct could be replace by an enum:
```rust
enum NetworkStream {
    PollingOneShot(oneshot::Receiver<mpsc::...>),
    PollingMpsc(mpsc::...),
}
```
showing the precise states of this stream.
But this is fine right now.

File: core/finality-grandpa/src/communication/gossip.rs [hunk 2]
```diff
@@ -183,7 +183,8 @@ impl<N: Ord> View<N> {
 
 const KEEP_RECENT_ROUNDS: usize = 3;
 
-/// Tracks topics we keep messages for.
+/// Tracks gossip topics we keep messages for. E.g. there is one topic per GRANDPA round and a global topic for commit
```
Comment (andresilva) [inline core/finality-grandpa/src/communication/gossip.rs]:
```suggestion
/// Tracks gossip topics that we are keeping messages for. We keep topics of the
```

File: core/finality-grandpa/src/communication/gossip.rs [hunk 3]
```diff
@@ -183,7 +183,8 @@ impl<N: Ord> View<N> {
...
+/// and catch-up messages.
```
Comment (andresilva) [inline core/finality-grandpa/src/communication/gossip.rs]:
```suggestion
/// last `KEEP_RECENT_ROUNDS` completed GRANDPA rounds, the topic for the
/// current and next round, and a global topic for commit and catch-up messages.
```

File: core/finality-grandpa/src/communication/gossip.rs [hunk 4]
```diff
@@ -256,7 +257,7 @@ fn neighbor_topics<B: BlockT>(view: &View<NumberFor<B>>) -> Vec<B::Hash> {
 #[derive(Debug, Encode, Decode)]
 pub(super) enum GossipMessage<Block: BlockT> {
 	/// Grandpa message with round and set info.
-	VoteOrPrecommit(VoteOrPrecommitMessage<Block>),
+	PrevoteOrPrecommit(PrevoteOrPrecommitMessage<Block>),
```
Comment (andresilva) [inline core/finality-grandpa/src/communication/gossip.rs]:
```suggestion
	Vote(VoteMessage<Block>),
```

WDYT?

File: core/finality-grandpa/src/communication/gossip.rs [hunk 5]
```diff
@@ -273,9 +274,9 @@ impl<Block: BlockT> From<NeighborPacket<NumberFor<Block>>> for GossipMessage<Blo
 	}
 }
 
-/// Network level message with topic information.
+/// Network level vote message with topic information.
 #[derive(Debug, Encode, Decode)]
-pub(super) struct VoteOrPrecommitMessage<Block: BlockT> {
+pub(super) struct PrevoteOrPrecommitMessage<Block: BlockT> {
```
Comment (andresilva) [inline core/finality-grandpa/src/communication/gossip.rs]:
```suggestion
pub(super) struct VoteMessage<Block: BlockT> {
```

File: core/finality-grandpa/src/communication/mod.rs [hunk 6]
```diff
@@ -707,7 +725,7 @@ impl<Block: BlockT, N: Network<Block>> Sink for OutgoingMessages<Block, N>
 				id: local_id.clone(),
 			};
 
-			let message = GossipMessage::VoteOrPrecommit(VoteOrPrecommitMessage::<Block> {
+			let message = GossipMessage::PrevoteOrPrecommit(PrevoteOrPrecommitMessage::<Block> {
```
Comment (andresilva) [inline core/finality-grandpa/src/communication/mod.rs]:
```suggestion
			let message = GossipMessage::Vote(VoteMessage::<Block> {
```

File: core/finality-grandpa/src/communication/mod.rs [hunk 7]
```diff
@@ -379,7 +397,7 @@ impl<B: BlockT, N: Network<B>> NetworkBridge<B, N> {
 			})
 			.and_then(move |msg| {
 				match msg {
-					GossipMessage::VoteOrPrecommit(msg) => {
+					GossipMessage::PrevoteOrPrecommit(msg) => {
```
Comment (andresilva) [inline core/finality-grandpa/src/communication/mod.rs]:
```suggestion
					GossipMessage::Vote(msg) => {
```

File: core/finality-grandpa/src/communication/mod.rs [hunk 8]
```diff
@@ -275,8 +292,8 @@ impl<B: BlockT, N: Network<B>> NetworkBridge<B, N> {
 				validator.note_round(Round(round.number), |_, _| {});
 
 				for signed in round.votes.iter() {
-					let message = gossip::GossipMessage::VoteOrPrecommit(
-						gossip::VoteOrPrecommitMessage::<B> {
+					let message = gossip::GossipMessage::PrevoteOrPrecommit(
```
Comment (andresilva) [inline core/finality-grandpa/src/communication/mod.rs]:
```suggestion
					let message = gossip::GossipMessage::Vote(
```

File: core/finality-grandpa/src/communication/mod.rs [hunk 9]
```diff
@@ -275,8 +292,8 @@ impl<B: BlockT, N: Network<B>> NetworkBridge<B, N> {
...
+						gossip::PrevoteOrPrecommitMessage::<B> {
```
Comment (andresilva) [inline core/finality-grandpa/src/communication/mod.rs]:
```suggestion
						gossip::VoteMessage::<B> {
```

File: core/finality-grandpa/src/communication/mod.rs [hunk 10]
```diff
@@ -48,7 +48,7 @@ use crate::{
 };
 use crate::environment::HasVoted;
 use gossip::{
-	GossipMessage, FullCatchUpMessage, FullCommitMessage, VoteOrPrecommitMessage, GossipValidator
+	GossipMessage, FullCatchUpMessage, FullCommitMessage, PrevoteOrPrecommitMessage, GossipValidator
```
Comment (andresilva) [inline core/finality-grandpa/src/communication/mod.rs]:
```suggestion
	GossipMessage, FullCatchUpMessage, FullCommitMessage, VoteMessage, GossipValidator
```

File: core/finality-grandpa/src/communication/gossip.rs [hunk 11]
```diff
@@ -1491,7 +1492,7 @@ mod tests {
 			}
 		});
 
-		let bad_sig = inner.validate_round_message(&peer, &VoteOrPrecommitMessage {
+		let bad_sig = inner.validate_round_message(&peer, &PrevoteOrPrecommitMessage {
```
Comment (andresilva) [inline core/finality-grandpa/src/communication/gossip.rs]:
```suggestion
		let bad_sig = inner.validate_round_message(&peer, &VoteMessage {
```

File: core/finality-grandpa/src/communication/gossip.rs [hunk 12]
```diff
@@ -1478,7 +1479,7 @@ mod tests {
 		val.note_round(Round(1), |_, _| {});
 
 		let inner = val.inner.read();
-		let unknown_voter = inner.validate_round_message(&peer, &VoteOrPrecommitMessage {
+		let unknown_voter = inner.validate_round_message(&peer, &PrevoteOrPrecommitMessage {
```
Comment (andresilva) [inline core/finality-grandpa/src/communication/gossip.rs]:
```suggestion
		let unknown_voter = inner.validate_round_message(&peer, &VoteMessage {
```

File: core/finality-grandpa/src/communication/gossip.rs [hunk 13]
```diff
@@ -612,7 +613,7 @@ impl<Block: BlockT> Inner<Block> {
 		cost::PAST_REJECTION
 	}
 
-	fn validate_round_message(&self, who: &PeerId, full: &VoteOrPrecommitMessage<Block>)
+	fn validate_round_message(&self, who: &PeerId, full: &PrevoteOrPrecommitMessage<Block>)
```
Comment (andresilva) [inline core/finality-grandpa/src/communication/gossip.rs]:
```suggestion
	fn validate_round_message(&self, who: &PeerId, full: &VoteMessage<Block>)
```

File: core/finality-grandpa/src/communication/gossip.rs [hunk 14]
```diff
@@ -1003,7 +1004,7 @@ impl<Block: BlockT> GossipValidator<Block> {
 
 		let action = {
 			match GossipMessage::<Block>::decode(&mut data) {
-				Ok(GossipMessage::VoteOrPrecommit(ref message))
+				Ok(GossipMessage::PrevoteOrPrecommit(ref message))
```
Comment (andresilva) [inline core/finality-grandpa/src/communication/gossip.rs]:
```suggestion
				Ok(GossipMessage::Vote(ref message))
```

File: core/finality-grandpa/src/communication/gossip.rs [hunk 15]
```diff
@@ -1163,7 +1164,7 @@ impl<Block: BlockT> network_gossip::Validator<Block> for GossipValidator<Block>
 				Ok(GossipMessage::Neighbor(_)) => false,
 				Ok(GossipMessage::CatchUpRequest(_)) => false,
 				Ok(GossipMessage::CatchUp(_)) => false,
-				Ok(GossipMessage::VoteOrPrecommit(_)) => false, // should not be the case.
+				Ok(GossipMessage::PrevoteOrPrecommit(_)) => false, // should not be the case.
```
Comment (andresilva) [inline core/finality-grandpa/src/communication/gossip.rs]:
```suggestion
				Ok(GossipMessage::Vote(_)) => false, // should not be the case.
```

**Comment by mxinden** [comment]
@andresilva @thiolliere thanks for the feedback. `PrevotePrecommit` -> `Vote` sounds great. In addition I like the idea of making `NetworkStream` an enum, making the two states of the state machine more obvious.

The recent 3 commits should address all comments. Can you take another look?

**Pull request merged**